sources/packages/backend/apps/workers/src/controllers/application/_tests_/e2e/application.controller-applicationChangeRequestApproval.e2e-spec.ts

```diff
@@ -0,0 +1,205 @@
+import {
+  ApplicationEditStatus,
+  ApplicationStatus,
+  Student,
+  User,
+} from "@sims/sims-db";
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeApplication,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import {
+  createFakeWorkerJob,
+  FAKE_WORKER_JOB_RESULT_PROPERTY,
+  FakeWorkerJobResult,
+  MockedZeebeJobResult,
+} from "../../../../../test/utils/worker-job-mock";
+import { createTestingAppModule } from "../../../../../test/helpers";
+import { ApplicationController } from "../../application.controller";
+import {
+  ApplicationChangeRequestApprovalJobInDTO,
+  ApplicationChangeRequestApprovalJobOutDTO,
+} from "../../application.dto";
+import { ICustomHeaders } from "@camunda8/sdk/dist/zeebe/types";
+import { createFakeApplicationChangeRequestApprovalPayload } from "./application-change-request-approval";
+import { SystemUsersService } from "@sims/services";
+
+describe("ApplicationController(e2e)-applicationChangeRequestApproval", () => {
+  let db: E2EDataSources;
+  let applicationController: ApplicationController;
+  let systemUser: User;
+  let sharedStudent: Student;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    db = createE2EDataSources(dataSource);
+    applicationController = nestApplication.get(ApplicationController);
+    sharedStudent = await saveFakeStudent(db.dataSource);
+    systemUser = nestApplication.get(SystemUsersService).systemUser;
+  });
+
+  it(`Should update application edit status to ${ApplicationEditStatus.ChangePendingApproval} when the Ministry approval is required.`, async () => {
+    // Arrange
+    const savedApplication = await saveFakeApplication(
+      db.dataSource,
+      { student: sharedStudent },
+      {
+        applicationStatus: ApplicationStatus.Edited,
+        applicationEditStatus: ApplicationEditStatus.ChangeInProgress,
+      },
+    );
+    const updateApplicationStatusPayload =
+      createFakeApplicationChangeRequestApprovalPayload(savedApplication.id);
+
+    // Act
+    const result = await applicationController.applicationChangeRequestApproval(
+      createFakeWorkerJob<
+        ApplicationChangeRequestApprovalJobInDTO,
+        ICustomHeaders,
+        ApplicationChangeRequestApprovalJobOutDTO
+      >(updateApplicationStatusPayload),
+    );
+
+    // Asserts
```

Comment:
Minor. should start calling `//Assert` as `//Asserts` on multiple assertions ?

Reply:
Copy and paste from other E2E tests 😉

Reply:
Changed.